specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/applications.json

```diff
@@ -233,7 +233,14 @@
         "accessModes": {
           "type": "array",
           "items": {
-            "type": "string"
+            "type": "string",
+            "enum": [
+              "WebPage"
+            ],
+            "x-ms-enum": {
+              "name": "ApplicationHttpsEndpointAccessMode",
+              "modelAsString": true
+            }
```

Comment:
This is a breaking change. Also a bit confused why do we have preview-api-version under stable directory?

Reply:
> This is a breaking change. Also a bit confused why do we have preview-api-version under stable directory?

@anuchandy  This is intentional. It’s a little confusing, but per rule #5 here, whether the swagger spec is in the preview or stable folder “is not a direct analog for whether or not an API Version has the ‘-preview’ suffix or not. SDKs that are generated from 'preview' folder items should indicate to their customers in the most idiomatic way that breaking changes may still be coming.”

Basically the short version is that HDInsight is a GA service that still has an API with a -preview tag. Somebody made the decision several years ago to release stable versions of Hydra/Hyak SDKs for .NET, JavaScript, etc. that consumed this the -preview API. We are now releasing stable swagger-based SDKs so we can at least finally deprecate the Hydra/Hyak SDKs. As a result, we are making a commitment that we will not make breaking changes to this swagger spec version so we can GA some swagger-based SDKs.

Reply:
@aim-for-better to address the "is this a breaking change" question - as per my comment on the other enum change, please confirm that WebPage is the only actual value the server can accept and return, and I think we're good.